### PR TITLE
Updated PR template and moved it to where GitHub would recognise it

### PR DESCRIPTION
**Pull request by lnewson**
Currently, a number of common things are forgotten and this is used by other teams, so I thought it might help us as well. If anyone can think of something extra to add, let me know.

### CODE REVIEW

**Comment by TheSpyder** [comment]
According to the docs where it was should be fine 🤔 
https://help.github.com/en/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository

The template changes will work for _us_, but maybe we should have a separate template for ourselves so community PRs aren't gated by our template requirements.

**Comment by lnewson** [comment]
> According to the docs where it was should be fine 
> https://help.github.com/en/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository
> 
> The template changes will work for _us_, but maybe we should have a separate template for ourselves so community PRs aren't gated by our template requirements.

If you read the link in detail, there's a small note where you need to use a query parameter to trigger pull request templates in the previous directory (specifically the `template` query parameter): https://help.github.com/en/github/managing-your-work-on-github/about-automation-for-issues-and-pull-requests-with-query-parameters

Eg: Here's what's needed to load the current template: https://github.com/tinymce/tinymce/compare/develop...TINY-3589?expand=1&template=pull_request_template.md

**Comment by lnewson** [comment]
Also we get very few community PR's similar to the docs, which also switched to this format. Regardless, the changes I've made will also work for community members, as it's basic things like change entries and tests.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Related Ticket: TINY-
+
+Description of Changes:
+* Placeholder text
+
+DOD:
```
Comment (metricjs) [inline .github/PULL_REQUEST_TEMPLATE.md:6]: DOD?

Comment (lnewson) [reply .github/PULL_REQUEST_TEMPLATE.md:6]: Copied that from the DX and Docs template tbh :shrug: I'm not sure what it actually means.

Comment (lnewson) [reply .github/PULL_REQUEST_TEMPLATE.md:6]: Ahh, it's `definition of done`

Comment (lnewson) [reply .github/PULL_REQUEST_TEMPLATE.md]: As discussed on slack, changed this to "Pre-checks" and added a couple other items

**Pull request merged**